### ISSUE TITLE
`code-block-tools`: tool availability check fails silently if `which` isn't installed

### ISSUE THREAD

**Issue by mcico** [rvben/rumdl#820]
### rumdl version

0.2.55

### Installation method

mise

### Bug description

`code-block-tools` reports a tool as "not found in PATH" even when it's installed and on `PATH`, if the `which` binary itself isn't installed.

`ToolExecutor::check_tool_exists()` checks availability by running `which <tool>`:

```rust
Command::new("which")
    .arg(tool_name)
    .stdout(Stdio::null())
    .stderr(Stdio::null())
    .status()
    .is_ok_and(|s| s.success())
```

If `which` can't spawn (`ENOENT`), `.status()` returns `Err`, and `.is_ok_and(...)` on an `Err` is `false` — same as the tool being absent. `execute()` then never even tries `Command::new(tool_name)` directly.

We hit this on a Fedora CI image without `which`. Every configured tool (`shellcheck`, `shfmt`, `tombi`) reported missing despite being correctly installed and on `PATH`; installing `which` fixed it with no other change.

### Steps to reproduce

1. On a system without `which`, install a tool that's genuinely on `PATH`, e.g. `shellcheck`.
2. Create `test.md`:

   `````markdown
   # Example

   ```bash
   echo "hello"
   ```
   `````
3. Configure:

   ```toml
   [code-block-tools]
   enabled = true

   [code-block-tools.languages]
   shell = { lint = ["shellcheck"] }
   ```
4. Run `rumdl check test.md`.

### Expected behavior

`shellcheck` runs against the shell block.

### Actual behavior

```text
test.md:1:1: [code-block-tools] tool 'shellcheck' not found in PATH
```
even though `command -v shellcheck` succeeds in that same shell.

### Configuration

```toml
[code-block-tools]
enabled = true
normalize-language = "linguist"
on-missing-language-definition = "fail"
on-missing-tool-binary = "fail-fast"

[code-block-tools.languages]
python = { lint = ["ruff:check"], format = ["ruff:format"] }
shell = { lint = ["shellcheck"], format = ["shfmt"] }
toml = { lint = ["tombi:lint"], format = ["tombi:format"] }
yaml = { format = ["yamlfmt"] }  # TODO: Consider adding ryl as linter
plaintext = { enabled = false }
text = { enabled = false }
```

### Operating system

Linux

### Additional context

As an idea, drop the dependency on an external `which` binary. Either try `Command::new(tool_name)` directly (e.g. `--version`) and treat `ENOENT` as "not found" — what `execute()` already does for the real call — or use a Rust-native PATH search like the `which` crate. Or alternatively fail with a message, that which is not present.

I analyzed this issue and wrote the bug report with the help of claude code  and have very little know-how about Rust. So please if that does not make much sense, just ignore it.

And btw. I really like rumdl 👍

**Comment by rvben** [comment]
Fixed in v0.2.56 (fb3e5d81, a005be4f). Tool availability is now resolved in-process by walking `PATH` the way `execvp` does (and, on Windows, the way `Command::new` resolves a name), so a missing `which` binary no longer reads as every tool being absent. Thanks for the report and the diagnosis, it pointed straight at the cause.
